lib/Doctrine/DBAL/Exception/ConstraintViolationException.php

```diff
@@ -28,4 +28,17 @@
  */
 class ConstraintViolationException extends ServerException
 {
+	/**
+	 * Get the name of affected column.
+	 *
+	 * @return string|NULL
+	 */
+	public function getAffectedColumnName()
+	{
+		if (preg_match_all('/^DETAIL\:\s+Key\s+\((\w+)\)/im', $this->getMessage(), $matches)) {
```

Comment:
@juniwalk relying on the exception message for such information is a bad idea (also consider different locale settings for exception messages etc.), thus I don't think we should add something like this to the core (especially as not all vendors provide that information).
This should be dealt with in userland with some kind of error handler class or similar if you really need it.